### DISCUSSION TITLE
ggcalendar revisit

### DISCUSSION THREAD

**Discussion by EvaMaeRey** [ggplot2-extenders/ggplot-extension-club#13] [Show and tell]
https://github.com/EvaMaeRey/ggcalendar is rewritten w/ readme2pkg (package-blog) style so all the details are in the readme. None of the functions are documented or exported at this point - moved a step back to minimal viable package.

You can answer the question, "What day is it?" 

![image](https://github.com/teunbrand/ggplot-extension-club/assets/10491910/9f74401a-cdc0-4804-a594-c9d645834072)

Another answer to that question from Winnie the Pooh, which is better.  "It's today" (Piglet) "My favorite day." (Pooh)  1926. AA Milne.  

We can also ask which day is the least traveled in the NYC flights data set in 2013?  It's Black Friday - the day after Thanksgiving.

![image](https://github.com/teunbrand/ggplot-extension-club/assets/10491910/d9bca08f-5387-4425-8856-1f51a25180fd)

**Comment by EvaMaeRey** [comment]
A comparative review of some calendar-motivated gg extensions with generous ggcalendar mention and extended discussion of author design goals for ggtilecal:  https://www.cynthiahqy.com/posts/ggplot-helper-design/

**Reply by teunbrand** [reply]
> Extending ggplot2 in this manner is not for the faint-hearted though. To do it well, you probably should read and fully understand the textbook, The Grammar of Graphics by Leland Wilkinson, and also have pretty advanced R programming skills.

Now I feel called out, as I've never read the book (and arguably just above-average R programming skills) :(

The author calls out an important point with regards to modularity of plotting helpers. It is why I loathe using these megaplot functions in e.g. {pheatmap} and their ilk.

**Reply by EvaMaeRey** [reply]
Time to read GoG!  Or just rely on ggplot2 being a faithful implementation and intuiting what was said?  Hoping we can hear more from the author!  https://fosstodon.org/@EvaMaeRey/112531435699954702 

I think a lot of the work that the gg calendar packages have in common is parsing dates into a dataframe w/ weekdays and week in month index and month.  I wonder if there's a designated function for this in lubridate or another dates-dedicated package.

**Reply by cynthiahqy** [reply]
@teunbrand I also have only skimmed the Grammar of Graphics textbook, so that comment was more aspirational than anything. Looking forward to joining the next meetup!

@EvaMaeRey there's a bit of a rabbit down to go down if you're interested in "calendrical calculations" -- e.g. https://github.com/espinielli/pycalcal?tab=readme-ov-file

**Reply by EvaMaeRey** [reply]
@cynthiahqy I left most of the 'calendrical calculations' (great term) to lubridate - definitely can't make all the world calendar flavors.  I actually just added the ability to switch to Monday start week days after.   Having different first day of the week that the local convention is obviously going to really be dissonant and unhelpful.  I recently learned Mandarin has day number in the name with Monday having '1' named in it while Portuguese Monday is segunda-feira (second day).

**Comment by teunbrand** [comment]
Unrelated to {ggcalendar} specifically, but can I ask what is up with the readme2pkg you appear to favour? Is it easier than just writing a package?

**Comment by EvaMaeRey** [comment]
I think it might be easier, but maybe especially for newer projects where you'd like to expose the underlying code and related rationale to your audience, and were ordering functionality helps make sense of it.   There are other projects that take this approach litr and fusen.  The 'readme2pkg' approach just uses the readme as the 'control document'.  The main support functionality is to move code from chunks into a file in a package folder.  I actually moved these functions to knitrExtra https://github.com/EvaMaeRey/knitrExtra.  The other projects, litr and fusen include templates as part of their packages.  Here's a discussion (w/ links to litr and fusen projects): https://evamaerey.github.io/mytidytuesday/2024-03-20-rladies-readme2pkg/rladies-readme2pkg-slides.html#1   As of last Friday, two other people have used knitrExtra + https://github.com/EvaMaeRey/readme2pkg.template.lite (making their first ever packages!) Yahoo!

**Reply by cynthiahqy** [reply]
I've used litr for similar reasons! Though, I haven't quite worked out when is the right time to move from litr to just a normal repository (to have access to `{usethis}` functions mostly).

**Reply by EvaMaeRey** [reply]
Oh cool!  I virtually met the litr authors at a an RLadies meeting.  Looks like are interested in knowing where it's being used. 
 Are you on this list? https://jacobbien.github.io/litr-project/articles/packages-in-the-wild.html
